src/compiler.ts

```diff
@@ -6290,6 +6339,28 @@ export class Compiler extends DiagnosticEmitter {
     }
   }
 
+  private injectClosedLocals(
+    expr: ExpressionRef,
+    type: Type
+  ): ExpressionRef {
+    var module = this.module;
+    var locals = type.locals!;
+    let _values = Map_values(locals);
+    var exprs = new Array<ExpressionRef>(_values.length);
+    for (let i = 0, k = _values.length; i < k; ++i) {
+      let local = unchecked(_values[i]);
+      let closureClass= type.classReference!
+      exprs[i] = module.store(
+        local.type.byteSize,
+        expr,
```

Comment:
This line appears to reuse the same `expr` multiple times instead of utilizing a temporary local. If IR nodes are not unique, this typically leads to all sorts of weird issues down the road (optimizing one modifies all with IR becoming invalid).

Reply:
Looks like I missed this one when you told me about this earlier- I'll do another scan through to find any other ones I may have missed